lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -30,9 +30,46 @@ def _config_str_to_list(self, list_str):
 
         return user_list
 
+    def _parse_app_id(self, exec_task):
+        app_id = ''
+        if 'app_id' in exec_task['job_input']:
+            app_id = exec_task['job_input']['app_id']
+            app_id = app_id.replace('.', '/')
+
+        return app_id
+
+    def _parse_method(self, exec_task):
+        method_id = ''
+        if 'method' in exec_task['job_input']:
+            method_id = exec_task['job_input']['method']
+            method_id = method_id.replace('/', '.')
+
+        return method_id
+
+    def _map_narrative(self, wsid, ws_narrs):
+        """
+        get the narrative name and version
+        """
+        n_name = ''
+        n_obj = '0'
+        for ws in ws_narrs:
+            if str(ws['workspace_id']) == str(wsid):
+                ws_name = ws['name']
+                n_name = ws_name
+                if not ws.get('meta', None) is None:
+                    w_meta = ws['meta']
+                    for w_m in w_meta:
```

Comment:
You don't have to go over every w_meta, as soon as you find a target just break. 

Also, I am not quite sure what you are trying to do with each w_m. You probably should add more test cases so that people know your logic. n_name can be assigned before and assigned once here? n_obj and n_name can be from different w_m?